internal/app/server/server.go

```diff
@@ -36,6 +40,13 @@ func Run(bootstrapConfig *bootstrapv1.Bootstrap,
 	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Should you be passing this ctx and cancel pair to the shutdown handler?

Reply:
right. will do that.

Reply:
actually there;s a `defer cancel()` already. https://github.com/envoyproxy/xds-relay/blob/master/internal/app/server/server.go#L37
We need not cancel the ctx in shutdown handler.

Reply:
In this case, can you get rid of the above TODO (I know you didn't add it)? I added it initially thinking that the function may not exit properly and cancel never gets called.

Reply:
i thought some more and figured there's value in cancelling the ctx anyway. Thanks for suggesting.